### ISSUE TITLE
Proposal: Use harden-runner in jobs using OPENTELEMETRYBOT_GITHUB_TOKEN

### ISSUE THREAD

**Issue by pellared** [open-telemetry/community#1556]
We can consider using https://github.com/step-security/harden-runner in places where `OPENTELEMETRYBOT_GITHUB_TOKEN` is used.

Proposal:
1. Try using  in one repository (e.g. in https://github.com/open-telemetry/opentelemetry-go/blob/main/.github/workflows/create-dependabot-pr.yml)
2. Propose a PR for https://github.com/open-telemetry/community/blob/main/assets.md#opentelemetry-bot to recommend using https://github.com/step-security/harden-runner
3. Create issues for repos which are using `OPENTELEMETRYBOT_GITHUB_TOKEN` so that they add https://github.com/step-security/harden-runner

**Comment by pellared** [comment]
CC @trask

**Comment by trask** [comment]
> We can consider using https://github.com/step-security/harden-runner in places where `OPENTELEMETRYBOT_GITHUB_TOKEN` is used.

the `OPENTELEMETRYBOT_GITHUB_TOKEN` fine-grained PAT org secret will have very limited permissions once #1549 is implemented

do you mean specifically for the additional fine-grained PATs with write access to a single repo for those who want to grant write permission to @opentelemetrybot? (https://github.com/open-telemetry/community/issues/1503#issuecomment-1588193654)

**Comment by pellared** [comment]
> do you mean specifically for the additional fine-grained PATs with write access to a single repo for those who want to grant write permission to @opentelemetrybot? 

Yup. However, it may be safer to use it everywhere as the permissions of the PAT may change and a contributor may not know what are the permissions.

**Comment by tigrannajaryan** [comment]
@pellared this fell through the cracks. Is it is still actual?

**Comment by pellared** [comment]
It is and I think that the proposal should be reviewed by Security SIG.

**Comment by pellared** [comment]
@tigrannajaryan, yes it is. I think that the proposal should be reviewed by Security SIG.

**Comment by trask** [comment]
cc @open-telemetry/sig-security-maintainers
